api/lib/domain/usecases/correct-answer-then-update-assessment.js

```diff
@@ -153,17 +161,25 @@ const correctAnswerThenUpdateAssessment = async function ({
 
   const challenge = await challengeRepository.get(answer.challengeId);
 
-  const onGoingCertificationChallengeLiveAlert =
-    await certificationChallengeLiveAlertRepository.getOngoingByChallengeIdAndAssessmentId({
-      challengeId: challenge.id,
+  let certificationCandidate;
+
+  if (assessment.isCertification()) {
+    const onGoingCertificationChallengeLiveAlert =
+      await certificationChallengeLiveAlertRepository.getOngoingByChallengeIdAndAssessmentId({
+        challengeId: challenge.id,
+        assessmentId: assessment.id,
+      });
+
+    if (onGoingCertificationChallengeLiveAlert) {
+      throw new ForbiddenAccess('An alert has been set.');
+    }
+
+    certificationCandidate = await certificationEvaluationCandidateRepository.findByAssessmentId({
```

Comment:
Je trouve que ce usecase prend un tres grande taille, je propose de decouper et isoler en fonction.

Ma proposition


```suggestion
  const certificationCandidate = await getEvaluatedCandidate({
    assessment,
    challenge,
    certificationChallengeLiveAlertRepository,
    certificationEvaluationCandidateRepository,
  });
```

Et du coup ma proposition de fonction

```js
const getEvaluatedCandidate = async ({
  assessment,
  challenge,
  certificationChallengeLiveAlertRepository,
  certificationEvaluationCandidateRepository,
}) => {
  if (assessment.isCertification()) {
    return;
  }

  const onGoingCertificationChallengeLiveAlert =
    await certificationChallengeLiveAlertRepository.getOngoingByChallengeIdAndAssessmentId({
      challengeId: challenge.id,
      assessmentId: assessment.id,
    });

  if (onGoingCertificationChallengeLiveAlert) {
    throw new ForbiddenAccess('An alert has been set.');
  }

  return certificationEvaluationCandidateRepository.findByAssessmentId({
    assessmentId: assessment.id,
  });
};
```